pkg/core/mpt/base.go

```diff
@@ -86,17 +86,7 @@ func encodeBinaryAsChild(n Node, w *io.BinWriter) {
 
 // encodeNodeWithType encodes node together with it's type.
 func encodeNodeWithType(n Node, w *io.BinWriter) {
-	switch t := n.Type(); t {
-	case HashT:
-		hn := n.(*HashNode)
-		if !hn.hashValid {
-			w.WriteB(byte(EmptyT))
-			break
-		}
-		fallthrough
-	default:
-		w.WriteB(byte(t))
-	}
+	w.WriteB(byte(n.Type()))
```

Comment:
We also need to adjust `(b *BaseNode) updateHash(n Node)` in the following manner:
```
if n.Type() == HashT || n.Type() == EmptyT {
    panic("can't update hash for hash or empty node")
}
```
Although it's hardly ever possible that this method will be called for an empty node. This is needed because previously `n.Type()` returned `HashT`  for an empty node.

Reply:
Still there